library/cohesity_oracle.py

```diff
@@ -0,0 +1,415 @@
+        prot_sources["environment"] = "Oracle"
+        current_status = get__protection_source_registration__status(
+            module, prot_sources)
+        check_mode_results = dict(
+            changed=False,
+            msg="Check Mode: Cohesity Protection Source is not currently registered",
+            id="")
+        if module.params.get('state') == "present":
+            if current_status:
+                check_mode_results[
+                    'msg'] = "Check Mode: Cohesity Protection Source is currently registered.  No changes"
+            else:
+                check_mode_results[
+                    'msg'] = "Check Mode: Cohesity Protection Source is not currently registered.  This action would register the Protection Source."
+                check_mode_results['id'] = current_status
+        else:
+            if current_status:
+                check_mode_results[
+                    'msg'] = "Check Mode: Cohesity Protection Source is currently registered.  This action would unregister the Protection Source."
+                check_mode_results['id'] = current_status
+            else:
+                check_mode_results[
+                    'msg'] = "Check Mode: Cohesity Protection Source is not currently registered.  No changes."
+        module.exit_json(**check_mode_results)
+
+    elif module.params.get('state') == "present":
+
+        if current_status:
+            prot_sources = dict(
+                token=get__cohesity_auth__token(module),
+                endpoint=module.params.get('endpoint'),
+                environment="Oracle"
+            )
+            oracle_status = get__protection_source_registration__status(module, prot_sources)
+            if not oracle_status:
+                resp = register_oracle_source(module, prot_sources, current_status)
+                if resp == True:
+                    results = dict(
+                        changed=True,
+                        msg="Registration of Cohesity Protection Source Complete")
+                else:
+                    results = dict(
+                        changed=False, msg=resp)
+                        #msg="Error while registering Cohesity Protection Source")
+
+            else:
+                results = dict(
+                    changed=False,
+                    msg="The Protection Source for this host is already registered",
+                    id=current_status,
+                    endpoint=module.params.get('endpoint'))
+        else:
+            sleep_count = 0
+
+            # Register the endpoint as Physical source.
+            response = register_source(module, prot_sources)
+
+            # Wait untill Physical source is successfully registered.
+            while sleep_count < 5:
+                sleep_count += 1
+                status = get__protection_source_registration__status(
+                    module, dict(environment="Physical", token=prot_sources['token'], endpoint=prot_sources['endpoint']))
```

Comment:
why is the environment Physical

Reply:
To register a source as Oracle, first the source must be registered as Physical source.